src/PrivateKey.h

```diff
@@ -17,30 +17,35 @@ class PrivateKey {
   public:
     /// The number of bytes in a private key.
     static const size_t size = 32;
-    /// The number of bytes in an extended private key.
-    static const size_t extendedSize = 3 * 32;
+    /// The number of bytes in a double extended key (used by Cardano)
+    static const size_t doubleExtendedSize = 2 * 3 * 32;
 
     /// The private key bytes.
     Data bytes;
-    /// Optional extended part of the key (additional 32 bytes)
-    Data extensionBytes;
-    /// Optional chain code (additional 32 bytes)
-    Data chainCodeBytes;
+
+    /// Optional members for extended keys and second extended keys
+    Data extension;
+    Data chainCode;
+    Data second;
+    Data secondExtension;
+    Data secondChainCode;
```

Comment:
Does it make sense to have a `CardanoPrivateKey` or (other naming) sub class?

Reply:
PrivateKey & PublicKey design is that one class supports different variants, with type as a member.  I would like using subclasses, but it adds some clutter, and it should be done consistently for both PrivateKey and PublicKey.